File: onebusaway-nyc-admin-webapp/src/main/webapp/js/oba/user-api-key-data.js
/*
 * Copyright (C)  2011 Metropolitan Transportation Authority
 *
 *  Licensed under the Apache License, Version 2.0 (the "License");
 *  you may not use this file except in compliance with the License.
 *  You may obtain a copy of the License at
 *
 *          http://www.apache.org/licenses/LICENSE-2.0
 *
 *  Unless required by applicable law or agreed to in writing, software
 *  distributed under the License is distributed on an "AS IS" BASIS,
 *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *  See the License for the specific language governing permissions and
 *  limitations under the License.
 */

$(document).ready( function () {
    var CONTEXT_PATH = $('#contextPathHolder').attr('data-contextPath');
    $('#userApiKeyData').DataTable({
        "lengthMenu": [[100, 500, 1000, -1], [100, 500, 1000, "All"]],
        ajax: {
            url: CONTEXT_PATH + '/api/api-key/data',
            dataSrc: ''
        },
        columns: [
            { data: 'created' },
            { data: 'name' },
            { data: 'email' },
            { data: 'projectName' },
            { data: 'projectUrl' },
            { data: 'platform' },
            { data: 'apiKey' },
        ]
    } );
} );

File: onebusaway-nyc-admin-webapp/src/main/java/org/onebusaway/nyc/admin/model/UserApiKeyData.java
/*
 * Copyright (C)  2011 Metropolitan Transportation Authority
 *
 *  Licensed under the Apache License, Version 2.0 (the "License");
 *  you may not use this file except in compliance with the License.
 *  You may obtain a copy of the License at
 *
 *          http://www.apache.org/licenses/LICENSE-2.0
 *
 *  Unless required by applicable law or agreed to in writing, software
 *  distributed under the License is distributed on an "AS IS" BASIS,
 *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *  See the License for the specific language governing permissions and
 *  limitations under the License.
 */

package org.onebusaway.nyc.admin.model;
import org.hibernate.annotations.GenericGenerator;
import org.hibernate.annotations.Index;
import javax.persistence.*;
import java.util.Date;

@Entity
@Table(name = "obanyc_api_key_data")
public class UserApiKeyData {
    @Id
    @GeneratedValue(strategy= GenerationType.AUTO, generator="native")
    @GenericGenerator(name = "native", strategy = "native")
    private Long id;

    @Column(name="name", nullable = false)
    private String name;

    @Column(name="created")
    @Temporal(TemporalType.TIMESTAMP)
    private Date created;

    @Column(name="email", nullable = false, length=512)
    private String email;

    @Column(name="project_name")
    private String projectName;

    @Column(name="project_url", length=512)
    private String projectUrl;

    @Column(name="platform")
    private String platform;

    @Column(name="api_key")
    @Index(name = "api_key_index")
    private String apiKey;

    @Column(name="comment", length=1024)
    private String comment;

    @PrePersist
    protected void onCreate() {
        created = new Date();
    }

    public Long getId() {
        return id;
    }

    public Date getCreated() {
        return created;
    }

    public void setCreated(Date created) {
        this.created = created;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public String getProjectName() {
        return projectName;
    }

    public void setProjectName(String projectName) {
        this.projectName = projectName;
    }

    public String getProjectUrl() {
        return projectUrl;
    }

    public void setProjectUrl(String projectUrl) {
        this.projectUrl = projectUrl;
    }

    public String getPlatform() {
        return platform;
    }

    public void setPlatform(String platform) {
        this.platform = platform;
    }

    public String getApiKey() {
        return apiKey;
    }

    public void setApiKey(String apiKey) {
        this.apiKey = apiKey;
    }

    public String getComment() {
        return comment;
    }

    public void setComment(String comment) {
        this.comment = comment;
    }


}
